src/Driver.cpp

```diff
@@ -184,3 +165,228 @@ void Driver::startAcquisition()
+                                                      + std::to_string(parity)
+                                                      + std::to_string(stopBits)
+                                                      + "\n").c_str()), 6);
+    readConfigurationAck();
+}
+
+void Driver::setFlowControlSettings(bool automaticEnsembleCycling,
+                                    bool automaticPingCycling,
+                                    bool binaryDataOutput,
+                                    bool enableSerialOutput,
+                                    bool enableDataRecording)
+{
+    setConfigurationMode();
+    writePacket(reinterpret_cast<uint8_t const*>(("CF" + std::to_string(automaticEnsembleCycling)
+                                                      + std::to_string(automaticPingCycling)
+                                                      + std::to_string(binaryDataOutput)
+                                                      + std::to_string(enableSerialOutput)
+                                                      + std::to_string(enableDataRecording)
+                                                      + "\n").c_str()), 8);
+    readConfigurationAck();
+}
+
+void Driver::setHeadingAlignment(int headingAlignment)
+{
+    sendStandardCommand(std::string("EA"), headingAlignment, 5, true);
+}
+
+void Driver::set_e_headingBias(int e_headingBias)
+{
+    sendStandardCommand(std::string("#EV"), e_headingBias, 5, true, true);
+}
+
+void Driver::setSalinity(int salinity)
+{
+    sendStandardCommand(std::string("ES"), salinity, 2);
+}
+
+void Driver::setSensorSourceSettings(SENSORSOURCE speedOfSoundSource,
+                                     SENSORSOURCE depthSource,
+                                     SENSORSOURCE headingSource,
+                                     SENSORSOURCE pitchAndRollSource,
+                                     SENSORSOURCE salinitySource,
+                                     SENSORSOURCE temperatureSource)
+{
+    setConfigurationMode();
+    writePacket(reinterpret_cast<uint8_t const*>(("EZ" + std::to_string(speedOfSoundSource)
+                                                      + std::to_string(depthSource)
+                                                      + std::to_string(headingSource)
+                                                      + std::to_string(pitchAndRollSource)
+                                                      + std::to_string(salinitySource)
+                                                      + std::to_string(temperatureSource)
+                                                      + "\n").c_str()), 7);
+    readConfigurationAck();
+}
+
+void Driver::setTimePerEnsemble(const base::Time& timePerEnsemble)
+{
+    setConfigurationMode();
+    std::vector<int> timeSplits = splitTime(timePerEnsemble);
+    std::vector<std::string> timeStrings;
+
+    for (int i = 0; i < 4; ++i)
+    {
+        std::string timeString;
+        if (getNumberOfDigits(i) == 1)
```

Comment:
You probably intended to check `getNumberOfDigits(itimeSplits.at(i)) == 1` here right?